conans/client/conan_api.py

```diff
@@ -68,7 +68,23 @@
 default_manifest_folder = '.conan_manifests'
 
 
+def api_method_hook(f):
```

Comment:
There is already a wrapper for the api call, I would prefer applying hooks in that one (a bit more "flat" call graph), instead of a dedicated wrapper for each thing.

Also, I would be minimal and introduce only the ``pre_command`` hook. We know there is a use case: do a ``conan config install``. I cannot think of anything that makes sense for a ``post_command``

Reply:
`post_command` could be used to manipulate logs, like, upload a log after creating a package

Reply:
We have specialized post_hooks for that though.

Reply:
On the nesting moment - it helps to elegantly special case one of the search command's api calls. It would be two same checks before and after the call otherwise which is a little bit more wasteful.